man/docker-build.1.md

```diff
@@ -51,6 +52,24 @@ cloned locally and then sent as the context.
    the remote context. In all cases, the file must be within the build context.
    The default is *Dockerfile*.
 
+**--build-arg**=*variable*
+   Set value for build-time variable. This option allows you to specify
+values of the variables that are available for expansion/substitution in the
+Dockerfile instructions like ADD, COPY etc, without an explicit prior definition by
+the ENV instruction. The build-time variables are also passed as environment
+context for the command(s) that will be executed as part of RUN instruction
+of Dockerfile, if there is no explicit prior definition by the ENV instruction.
+Normally, these variables are not persisted in the resulting Docker image. This gives
+the flexibility to build an image by passing host specific environment variables (like
+http_proxy) that will be used on the RUN commands without affecting portability
+of the generated image.
+However, as with any variable, they can be persisted in the final image if they are used in an
+ENV instruction (e.g. ENV myName=$myName will save myName in the image).
+
+Only the build-time variables that are defined using the ARG instruction of Dockerfile
+are allowed to be expanded or passed as environment to the RUN command. Read more about
+ARG instruction in Dockerfile reference.
+
```

Comment:
yes, it is documented in the ARG's reference, that the man page refers..

https://github.com/docker/docker/pull/15182/files#diff-4fc21624aa5dadd153e2780910da7c7dR1014